lab-steven/server.js

```diff
@@ -0,0 +1,110 @@
+'use strict';
+
+const net = require('net');
+const EE = require('events');
+const Client = require('./model/client-constructor.js');
+const PORT = process.env.PORT || 8080;
+const server = net.createServer();
+const ee = new EE();
+
+const connectedClients = [];
+
+ee.on('@all', function(user, data) {
+  connectedClients.forEach(function(client) {
+    client.socket.write(`${user.nickname}: ${data}\r\n`);
+  });
+});
+
+ee.on('@nickname', function(user, data) {
+  user.nickname = data.trim();
+  user.socket.write(`Changed nickname to ${data}\r\n`);
+});
+
+ee.on('@pm', function(user, data) {
+  var message = data.split(' ').slice(1).join(' ').trim();
+  var targetUser = data.split(' ').shift().trim();
+
+  var foundUser = connectedClients.filter(function(client) {
+    return (client.nickname === targetUser || client.id === targetUser);
+  })[0];
+
+  if (foundUser) {
+    user.socket.write(`To ${foundUser.nickname}: ${message}\r\r\n\n`);
```

Comment:
Your `\r\n` should be in pairs when you need more the one of them I.E. `\r\n\r\n`